cmd/swarmctl/service/logs.go

```diff
@@ -45,13 +53,21 @@ var (
 				serviceIDs = append(serviceIDs, service.ID)
 			}
 
+			stdstreams := []api.LogStream{}
+			if !noStdout {
+				stdstreams = append(stdstreams, api.LogStreamStdout)
+			}
+			if !noStderr {
+				stdstreams = append(stdstreams, api.LogStreamStderr)
+			}
```

Comment:
Actual behavior is defaults for empty field in the API call, so you get both. It should probably return an error though.